### PR TITLE
Introduce Sign and Verify classes

### PR DESCRIPTION
**Pull request by twe4ked**
This is essentially a copy paste the internal code hasn't changed up to ee140b7.

Past that I've made some minor tweaks. Probably best to review individual commits.

### CODE REVIEW

**Review by vonconrad — APPROVED** [review]
Nice refactor!

**Review by warrenseen — APPROVED** [review]
Perfect! 💯

File: lib/jwt_signed_request/claims.rb [hunk 1]
```diff
@@ -51,6 +48,14 @@ def generate
 
     private_constant :DEFAULT_TIMEOUT
 
+    EMPTY_HEADERS = [].freeze
+
+    private_constant :EMPTY_HEADERS
+
+    def timeout
+      DEFAULT_TIMEOUT
+    end
```
Comment (yoshdog) [inline lib/jwt_signed_request/claims.rb:57]: i thought we wanted to give people the ability to set the timeouts?

**Review by yoshdog — APPROVED** [review]
Looks good..
prob need the ability to allow people to set the timeouts when they sign requests. But we can add that in a new PR.

**Comment by twe4ked** [comment]
@yoshdog There was support for it at the Claims level but not from the public API. Can definitely be added in later though.

**Pull request merged**